odxtools/dataobjectproperty.py

```diff
@@ -136,11 +136,15 @@ def decode_from_pdu(self, decode_state: DecodeState) -> ParameterValue:
 
         if self.compu_method.is_valid_internal_value(internal):
             return self.compu_method.convert_internal_to_physical(internal)
-        else:
-            # TODO: How to prevent this?
-            raise DecodeError(
-                f"DOP {self.short_name} could not convert the coded value "
-                f" {repr(internal)} to physical type {self.physical_type.base_data_type}.")
+
+        default_value = getattr(self.compu_method.compu_internal_to_phys, "compu_default_value",
+                                None)
+        if default_value is not None:
+            return str(default_value)
+
+        # TODO: How to prevent this?
+        raise DecodeError(f"DOP {self.short_name} could not convert the coded value "
```

Comment:
@andlaus what do you think about using `str(internal_value)` when not in strict mode as final fallback ?

Reply:
I don't really know: my hunch is that `None` is better because if the callling code encounters an unexpected type like `str` instead of `int` it almost certainly goes belly-up anyway and it is probably mentally a bit easier to associate `None` with "did not work". On the other hand, the internal value of the parameter would be lost in the warnings...